src/NetMQ/ThreadSafe/ReceiveSocketExtensions.cs

```diff
@@ -0,0 +1,236 @@
+        public static bool TryReceiveString([NotNull] this IThreadSafeInSocket socket, [NotNull] Encoding encoding, out string str)
+        {
+            return socket.TryReceiveString(TimeSpan.Zero, encoding, out str);
+        }
+
+        #endregion
+
+        #region Timeout
+
+        /// <summary>
+        /// Attempt to receive a string from <paramref name="socket"/>, and decode using <see cref="SendReceiveConstants.DefaultEncoding"/>.
+        /// If no message is available within <paramref name="timeout"/>, return <c>false</c>.
+        /// </summary>
+        /// <param name="socket">The socket to receive from.</param>
+        /// <param name="timeout">The maximum period of time to wait for a message to become available.</param>
+        /// <param name="str">The content of the received message, or <c>null</c> if no message was available.</param>
+        /// <returns><c>true</c> if a message was available, otherwise <c>false</c>.</returns>
+        public static bool TryReceiveString([NotNull] this IThreadSafeInSocket socket, TimeSpan timeout, out string str)
+        {
+            return socket.TryReceiveString(timeout, SendReceiveConstants.DefaultEncoding, out str);
+        }
+
+        /// <summary>
+        /// Attempt to receive a string from <paramref name="socket"/>, and decode using <paramref name="encoding"/>.
+        /// If no message is available within <paramref name="timeout"/>, return <c>false</c>.
+        /// </summary>
+        /// <param name="socket">The socket to receive from.</param>
+        /// <param name="timeout">The maximum period of time to wait for a message to become available.</param>
+        /// <param name="encoding">The encoding used to convert the data to a string.</param>
+        /// <param name="str">The content of the received message, or <c>null</c> if no message was available.</param>
+        /// <returns><c>true</c> if a message was available, otherwise <c>false</c>.</returns>
+        public static bool TryReceiveString([NotNull] this IThreadSafeInSocket socket, TimeSpan timeout, [NotNull] Encoding encoding, out string str)
+        {
+            var msg = new Msg();
+            msg.InitEmpty();
+
+            if (socket.TryReceive(ref msg, timeout))
+            {
+                str = msg.Size > 0
+                    ? msg.GetString(encoding)
+                    : string.Empty;
+
+                msg.Close();
+                return true;
+            }
+
+            str = null;
+            msg.Close();
+            return false;
+        }
+
+        #endregion
+        
+        #region Async
+
+        /// <summary>
+        /// Receive a string from <paramref name="socket"/> asynchronously.
+        /// </summary>
+        /// <param name="socket">The socket to receive from.</param>
+        /// <returns>The content of the received message.</returns>
+        public static Task<string> ReceiveStringAsync([NotNull] this IThreadSafeInSocket socket)
```

Comment:
Cool, only issue is that it will only be supported on standard 2.1 and .net 4.7.
What do you think, to have here a condition or to change NetMQ to support standard 2.1 and .net 4.7?

Reply:
Nevermind, I have just found out about System.Threading.Tasks.Extensions